### PR TITLE
HistoryStore for tracking relevant routes

### PR DESCRIPTION
**Pull request by jacogr**
Split from https://github.com/ethcore/parity/pull/4178

- stores navigation entries in localstorage
- currently enabled for dapps & accounts
- /home uses these to display the recently-used

### CODE REVIEW

File: js/src/routes.js [hunk 1]
```diff
@@ -16,11 +16,15 @@
 
 import {
   Accounts, Account, Addresses, Address, Application,
-  Contract, Contracts, Dapp, Dapps,
+  Contract, Contracts, Dapp, Dapps, HistoryStore,
```
Comment (jacogr) [inline js/src/routes.js]: Not 100% convinced on this location. It is only related to keeping a history of views, but I'm open to suggestions to other locations.

Comment (ngotchac) [inline js/src/routes.js]: Would it be better in `~/util` ?

Comment (jacogr) [inline js/src/routes.js]: My knee-jerk reaction was that `~/util` is even worse, my thoughts after 30s is that it _may_ fit. (Although it really is meant for swiss-army-knife utilities)

Comment (ngotchac) [inline js/src/routes.js]: Yep that's what I think too but it's true that it feels a bit strange/out-of-place in `~/views`... @derhuerst any thoughts?

**Comment by ngotchac** [comment]
`mustntgrumble` only related to the `HistoryStore` path. Looks good otherwise !

**Comment by jacogr** [comment]
Leaving the path for now - I'm actually hoping another store ends up being the same that way will move it in `~/stores` or (probably better) refactor the `~/redux` stuff properly.

**Pull request merged**